travis.sh

```diff
@@ -69,16 +250,78 @@ function runFindbugs() {
   fi
 }
 
+##########################################################
+# Run CPD for the langauge passed, check for failures and
+# move the final result to a separate file.
+#
+# Arguments:
+#   arg1: Language for which CPD is run (Java or Scala)
+#   arg2: Duplicates threshold for the language
+#   arg3: Name of the threshold constant for the language
+#   arg4: List of files changed in the PR
+# Returns:
+#   None
+##########################################################
+function runCPDForLanguage() {
+  sbt cpd
+  if [ $? -ne 0 ]; then
+    echo -e "$ERROR_COLOR_PREFIX CPD for "$1" failed..."
+    exit 1;
+  fi
+  cpd_result_file=$(getCPDReportName $1)
+  mv $CPD_REPORT_PATH $cpd_result_file
+  if [ $1 = "Scala" ]; then
+    removeLicenseHeaderDuplicates $cpd_result_file
+  fi
+  checkCPDReport $1 $2 $3 $cpd_result_file "${4}"
+}
+
+##########################################################
+# Run CPD for Java and Scala one by one. For Scala, first
+# change cpdLanguage setting in cpd.sbt to Language.Scala
+# and then run CPD.
+#
+# Arguments:
+#   arg1: List of files changed in the PR
+# Returns:
+#   None
+##########################################################
+function runCPD() {
+  echo -e "$INFO_COLOR_PREFIX Running Copy Paste Detector(CPD) for Java..."
+  runCPDForLanguage "Java" $JAVA_CPD_THRESHOLD "JAVA_CPD_THRESHOLD" "${1}"
+
+  echo ""
+  # Change language to Scala before running CPD again.
+  sed -i '1,$s/Language.Java/Language.Scala/g' cpd.sbt
```

Comment:
Would be good to use cpdLanguage parameter to pass language as parameter, rather than making changes in cpd.sbt file.